### PR TITLE
Laboratory work #1, Zheleznova Tatiana - 22FPL2

### PR DESCRIPTION
**Pull request by tanya-iron**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -0,0 +1 @@
+print('Hello, world!')
```
Comment (artyomtugaryov) [inline main.py:1]: please, remove the file

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Unfortunately, nothing to review

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -6,11 +6,22 @@
...
+       removes punctuation, digits and other symbols
+       :param text: a text
+       :return: a list of lower-cased tokens without punctuation
+       """
+    if type(text) != str:
+        return None
+    text = text.lower()
+    tokens = []
+    for token in text:
+        if token.isalpha() and token != 'º':
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:19]: where did you find this symbol? And for what?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py:19]: > where did you find this symbol? And for what?

It was in the English text, it's the degree sign. It is considered a letter, so my code kept counting it.

File: main.py (hunk 1 above)
Comment (tanya-iron) [reply main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -19,6 +30,17 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
     :return: a dictionary with frequencies
     """
+    if type(tokens) != list:
+        return None
+    frequency_dict = {}
+    tokens_number = len(tokens)
+    for token in tokens:
+        if type(token) != str:
+            return None
+            break
```
[the remark this answers is not in the document]
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -28,6 +50,12 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if type(language) != str or type(text) != str:
```
[the remark this answers is not in the document]
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -37,6 +65,16 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
     :param actual: a list of actual values
     :return: the score
     """
+    if len(predicted) != len(actual):
+        return None
+    if type(predicted) != list or type(actual) != list:
```
[the remark this answers is not in the document]
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/start.py [hunk 6]
```diff
@@ -3,7 +3,8 @@
 """
 
 
-def main() -> None:
+def main():
+    from main import create_language_profile
```
[the remark this answers is not in the document]
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/start.py]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -2,15 +2,27 @@
...
 
 
 def tokenize(text: str) -> list[str] | None:
     """
-    Splits a text into tokens, converts the tokens into lowercase,
-    removes punctuation, digits and other symbols
-    :param text: a text
-    :return: a list of lower-cased tokens without punctuation
-    """
+       Splits a text into tokens, converts the tokens into lowercase,
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:10]: revert the string

Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:10]: revert the string

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -2,15 +2,27 @@
...
+       removes punctuation, digits and other symbols
+       :param text: a text
+       :return: a list of lower-cased tokens without punctuation
+       """
+    if not isinstance(text, str):
+        return None
+    text = text.lower()
+    tokens = []
+    for token in text:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:19]: list comprehension

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -19,6 +31,16 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list):
+        return None
+    frequency_dict = {}
+    tokens_number = len(tokens)
+    for token in tokens:
+        if not isinstance(token, str):
+            return None
+        token_frequency = tokens.count(token)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:41]: what if tokens contains a token not once - you count the same value multi times

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 9 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py:42]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 7 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -11,6 +12,14 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    text = text.lower()
+    tokens = [token for token in text if token.isalpha()]
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:18]: do not create the variable - return as is

File: lab_1_classify_by_unigrams/main.py [hunk 11]
```diff
@@ -11,6 +12,14 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    text = text.lower()
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:17]: why do you create this variable? make lower once at the loop

File: lab_1_classify_by_unigrams/main.py [hunk 12]
```diff
@@ -19,6 +28,20 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    if not isinstance(tokens, list):
+        return None
+    frequency_dict = {}
+    tokens_number = len(tokens)
+    token_list = []
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:35]: try `set`

File: lab_1_classify_by_unigrams/main.py [hunk 13]
```diff
@@ -28,6 +51,13 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
     :param text: a text
     :return: a dictionary with two keys – name, freq
     """
+    if not isinstance(language, str) or not isinstance(text, str):
+        return None
+    frequencies = calculate_frequencies(tokenize(text))
+    if isinstance(frequencies, dict):
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:57]: revert the condition and return None

File: lab_1_classify_by_unigrams/main.py [hunk 14]
```diff
@@ -37,6 +67,16 @@ def calculate_mse(predicted: list, actual: list) -> float | None:
...
     :return: the score
     """
+    if len(predicted) != len(actual):
+        return None
+    if not isinstance(predicted, list) or not isinstance(actual, list):
+        return None
+    mse_sum = 0
+    for index, element in enumerate(predicted):
+        mse_sum += (element - actual[index]) ** 2
+    mse = mse_sum / len(predicted)
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:77]: do not create the variable

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Very good!

File: lab_1_classify_by_unigrams/main.py (hunk 14 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 13 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 12 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 11 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 10 above)
Comment (tanya-iron) [reply lab_1_classify_by_unigrams/main.py]: fixed

**Pull request closed without merging**